superchain/superchain.go

```diff
@@ -132,16 +135,18 @@ func (c *ChainConfig) SetDefaultHardforkTimestampsToNil(s *SuperchainConfig) {
 }
 
 // setNilHardforkTimestampsToDefault overwrites each unspecified hardfork activation time override
-// with the superchain default.
+// with the superchain default, if the default is not nil and is after the SuperchainTime
 func (c *ChainConfig) setNilHardforkTimestampsToDefault(s *SuperchainConfig) {
 	cVal := reflect.ValueOf(&c.HardForkConfiguration).Elem()
 	sVal := reflect.ValueOf(&s.hardForkDefaults).Elem()
 
 	for i := 0; i < reflect.Indirect(cVal).NumField(); i++ {
 		overrideValue := cVal.Field(i)
-		if overrideValue.IsNil() {
-			defaultValue := sVal.Field(i)
-			overrideValue.Set(defaultValue)
+		defaultValue := sVal.Field(i)
+		if overrideValue.IsNil() && !defaultValue.IsNil() {
+			if c.SuperchainTime != nil && reflect.Indirect(defaultValue).Uint() >= *c.SuperchainTime {
```

Comment:
You can merge these two. And the inverse `c.SuperchainTime != nil` check you can actually move to the top of `setNilHardforkTimestampsToDefault` as an early exit if it's `nil`.

Reply:
[45f44ae](https://github.com/ethereum-optimism/superchain-registry/pull/233/commits/45f44ae5e44a2627d68d1b9ee11a798abaedbc83)